chat/message.py

```diff
@@ -125,29 +127,54 @@ def updateConversationStatus(self, conn) -> bool:
             raise Exception('no conversation record')
 
         cur = conn.execute('''
-            SELECT DISTINCT user_id
-            FROM %(schema_name)s.receipt
-            WHERE message_id = %(message_id)s AND read_at IS NOT NULL
+            WITH
+              read_count AS (
+                SELECT DISTINCT COUNT(user_id) as count
+                FROM %(schema_name)s.receipt
+                WHERE message_id = %(message_id)s
+                    AND read_at IS NOT NULL
+              ),
+              participant_count AS (
+                SELECT participant_count as count
+                FROM %(schema_name)s.conversation
+                WHERE _id = %(conversation_id)s
+              )
+            UPDATE %(schema_name)s.message
+            SET _updated_at = NOW(),
+                conversation_status =
+                  CASE
+                    WHEN read_count.count = 0 THEN 'delivered'
+                    WHEN read_count.count < participant_count.count
+                        THEN 'some_read'
+                    ELSE 'all_read'
+                  END
+            FROM read_count, participant_count
+            WHERE _id = %(message_id)s
+            RETURNING _updated_at, conversation_status
             ''', {
                 'schema_name': AsIs(_get_schema_name()),
-                'message_id': self.record.id.key
+                'message_id': self.record.id.key,
+                'conversation_id': self.conversationRecord.id.key
             }
         )
 
-        read_users = set([row[0] for row in cur if row[0]])
+        return_value = cur.fetchone()
+        if return_value is not None:
+            self.record['_updated_at'] = return_value[0]
+            self.record['conversation_status'] = return_value[1]
```

Comment:
actually you can use name such as `row["_updated_at"]`, but you are not required to change this now